### PR TITLE
Fix crash related to "delete all clusters", thermal & renewable

### LINKED ISSUE #297: Crash when creating a thermal / renewable cluster
**Description**
Crash when creating a thermal / renewable cluster in some situation

**How to Reproduce**
- Open a study containing clusters (thermal or renewable)
- Delete all clusters of an area
- Save the study
- Create a new cluster
- Crash occurs

**Expected behavior**
When creating the new cluster, Antares should not crash, it should create the cluster

**OS and version**
 - OS: Windows
 - Antares Simulator version: 8.0.0

### PR DESCRIPTION
**Pull request by flomnes**
Fix #297 

* `clusterCount` becomes a method
* Use `std::vector` instead of C-style arrays.

### CODE REVIEW

File: src/libs/antares/study/parts/renewable/container.h [hunk 1]
```diff
@@ -106,11 +106,13 @@ class PartRenewable
     ** This list is mainly used to ensure the same order of the
     ** renewable clusters in the outputs.
     */
-    RenewableCluster** clusters;
+    std::vector<RenewableCluster*> clusters;
     //! How many clusters have we got ?
     // Only available from the solver
-    uint clusterCount;
-
+    inline uint clusterCount() const
```
Comment (guilpier-code) [inline src/libs/antares/study/parts/renewable/container.h]: I have a warning here : 
`container.h(131,1): warning C4267: 'return': conversion from 'size_t' to 'uint', possible loss of data`

File: src/libs/antares/study/parts/thermal/container.h [hunk 2]
```diff
@@ -124,11 +124,12 @@ class PartThermal
...
     ** thermal clusters in the outputs.
     */
-    ThermalCluster** clusters;
-    //! How many clusters have we got ?
-    // Only available from the solver
-    uint clusterCount;
+    std::vector<ThermalCluster*> clusters;
 
+    // Return unit count
+    inline uint clusterCount() const {
```
Comment (guilpier-code) [inline src/libs/antares/study/parts/thermal/container.h]: `container.h(131,1): warning C4267: 'return': conversion from 'size_t' to 'uint', possible loss of data`

File: src/libs/antares/study/parts/renewable/container.cpp [hunk 3]
```diff
@@ -38,7 +38,7 @@ namespace Antares
 {
 namespace Data
 {
-PartRenewable::PartRenewable() : list(), clusters(nullptr), clusterCount((uint)-1)
+PartRenewable::PartRenewable() : list(), clusters()
```
Comment (guilpier-code) [inline src/libs/antares/study/parts/renewable/container.cpp]: Do we really need the call to default `clusters()` constructor ?

File: src/libs/antares/study/parts/thermal/container.cpp [hunk 4]
```diff
@@ -39,12 +39,7 @@ namespace Antares
 namespace Data
 {
 PartThermal::PartThermal() :
- unsuppliedEnergyCost(0.),
- spilledEnergyCost(0.),
- list(),
- mustrunList(),
- clusters(nullptr),
- clusterCount((uint)-1)
+ unsuppliedEnergyCost(0.), spilledEnergyCost(0.), list(), mustrunList(), clusters()
```
Comment (guilpier-code) [inline src/libs/antares/study/parts/thermal/container.cpp]: Same thing as above : 
Do we really need the call to default `clusters()` constructor ?

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/libs/antares/study/parts/renewable/container.cpp (hunk 3 above)
Comment (flomnes) [reply src/libs/antares/study/parts/renewable/container.cpp]: Right, I also removed the call to other default ctors.

File: src/libs/antares/study/parts/thermal/container.cpp (hunk 4 above)
Comment (flomnes) [reply src/libs/antares/study/parts/thermal/container.cpp]: No, I removed this call and others.

File: src/libs/antares/study/parts/thermal/container.h (hunk 2 above)
Comment (flomnes) [reply src/libs/antares/study/parts/thermal/container.h]: OK

File: src/libs/antares/study/parts/renewable/container.h (hunk 1 above)
Comment (flomnes) [reply src/libs/antares/study/parts/renewable/container.h]: OK

**Review by guilpier-code — APPROVED** [review]

**Pull request merged**